scylla/src/transport/caching_session.rs

```diff
@@ -250,7 +250,7 @@ mod tests {
 
         let mut rows = 0;
 
-        while let Some(_) = iter.next().await {
+        while iter.next().await.is_some() {
             rows += 1;
         }
```

Comment:
By the way, I think this doesn't check for errors. The return type of `iter.next()` is `Option<Result<Row>>`, and the inner result isn't checked.

How about this:

```rust
use futures::stream::TryStreamExt;
let rows = iter.try_collect::<Vec<_>>().await.unwrap().len();
```

Reply:
oh, and Option isn't `must_use`, sneaky. Good idea, I'll change accordingly

Reply:
ugh, it gets a little messy, because our RowIterator doesn't implement StreamExt (it only implements Stream), and thus doesn't have try_collect. I suggest pushing this change to a follow-up and keep this one about clippy only (since the problem with ignoring error results is preexisting).

Reply:
oh wait, maybe I'm just missing a `use` directive here

Reply:
done, thx

Reply:
That's very strange. StreamExt/TryStreamExt should be implemented automatically for RowIterator. I applied the same change on my local copy of the repo and it works.

All in all, I agree that this can be done in the followup.

Reply:
Oh, I see that you fixed it :)